### PR TITLE
Allow customizing the promotion code batch mailer class

### PR DESCRIPTION
**Pull request by jtapia**
Adds the ability to replace default promotion code batch mailer class with a custom one, so there's no need to override default partials

### CODE REVIEW

File: core/lib/spree/app_configuration.rb [hunk 1]
```diff
@@ -307,6 +307,15 @@ def default_pricing_options
     # @api experimental
     class_name_attribute :shipping_rate_tax_calculator_class, default: 'Spree::TaxCalculator::ShippingRate'
 
+    # Allows providing your own Mailer for order mailer.
```
Comment (kennyadsl) [inline core/lib/spree/app_configuration.rb]: This comment needs to be updated

File: core/lib/spree/app_configuration.rb [hunk 2]
```diff
@@ -307,6 +307,15 @@ def default_pricing_options
...
+    #
+    # @!attribute [rw] order_mailer_class
```
Comment (kennyadsl) [inline core/lib/spree/app_configuration.rb]: here as well

File: core/lib/spree/app_configuration.rb (hunk 1 above)
Comment (jtapia) [reply core/lib/spree/app_configuration.rb]: thanks, done

File: core/lib/spree/app_configuration.rb (hunk 2 above)
Comment (jtapia) [reply core/lib/spree/app_configuration.rb]: thanks, done

**Pull request merged**